tests/main_test.py

```diff
@@ -216,3 +216,19 @@ def test_result_as_dict():
                      'domain': 'google',
                      'suffix': 'com'}
     assert result._asdict() == expected_dict
+
+
+def test_punycode_extended():
+    """
+    "narrow python" builds may break on certain unicode characters
```

Comment:
Can you colocate this test with other punycode test(s)? Can you expand this comment how it differs from those other tests? 

CI passes with or without the library code change in this PR. Worth mentioning that? And what the test _should_ do on a narrow build? I'm fine with "should." For this edge case, I don't care to maintain a narrow build environment in CI. :blush: